x-pack/plugins/fleet/server/services/actions/client.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+
+import { FleetActionsClientError } from '../../../common/errors';
+
+import type {
+  FleetActionsClientInterface,
+  FleetActionRequest,
+  FleetActionResult,
+  BulkCreateResponse,
+} from './types';
+import {
+  createAction,
+  bulkCreateActions,
+  getActionsByIds,
+  getActionsWithKuery,
+  getActionResultsByIds,
+  getActionResultsWithKuery,
+} from './actions';
+
+export class FleetActionsClient implements FleetActionsClientInterface {
+  constructor(private esClient: ElasticsearchClient, private packageName: string) {
```

Comment:
The name `package` sounds out of place for actions, shouldn't it be called `pluginName` or `actionType`?

Reply:
I went with `packageName` as it is what we use some of the other clients (artifacts, files to name a few). It is actually the integration package name. I can change it if you insist, else I'll leave it as is for now.